### PR TITLE
Fix toolbar focus bug

### LINKED ISSUE #6835: fix: clicking button in toolbar temporarily focuses wrong element
# 🐛 Bug Report

In the toolbar, if you click a button that is not associated with the `activeIndex` of the toolbar, the `activeIndex` element temporarily gets focus before the clicked button gets focus. The component is in the temporary state while the mouse is pressed down (but not released) during the click on the button.

As a result, the toolbar can have an odd visualization where one button is focused but a different button is active.

As far as I can tell, the root cause of the issue is that the `focusin` handler on the toolbar runs before the `click` handler on the toolbar. This causes the `focusin` handler to focus the element associated with `this.activeIndex` in the toolbar until the `click` handler runs to update `this.activeIndex` and set that element to being focused.

## 💻 Repro or Code Sample
I can reproduce this in the fast-foundation toolbar storybook page by making the following changes:
1. set the `storyTemplate` to:
```html
<style>
    button:focus {
        background: red;
    }
    fast-toolbar {
        background: pink;
    }
</style>
<fast-toolbar
    orientation="${x => x.orientation}"
    :ariaLabel="${x => x.ariaLabel}"
    :ariaLabelledby="${x => x.ariaLabelledby}"
>
    ${x => x.storyContent}
</fast-toolbar>
<button>something else on the page</button>
```
2. set the `storyContent` to:
```
<button>Button1</button>
<button>Button2</button>
<button>Button3</button>
<select>
    <option>Option 1</option>
    <option>Second option</option>
    <option>Option 3</option>
</select>
<label for="check-1">
    <input type="checkbox" name="checkbox" id="check-1" />
    Checkbox 1
</label>
<label for="check-2">
    <input type="checkbox" name="checkbox" id="check-2" />
    Checkbox 2
</label>
<label for="check-3">
    <input type="checkbox" name="checkbox" id="check-3" />
    Checkbox 3
</label>
<input type="text" name="text" id="text-input" />
```
3. Click on a button in the toolbar, for example"Button2"
4. Click on the button outside of the toolbar
5. Click on a different button in the toolbar, for example "Button3"
6. Notice that while the mouse is being held down on "Button3", "Button2" incorrectly is in the focused state. "Button3" correctly enters the focused state once the mouse is released.

toolbar-focus-bug

## 🤔 Expected Behavior

The toolbar button being pressed should have focus, regardless of which item in the toolbar previously had focus.

## 😯 Current Behavior

The previously focused button keeps focus while a new button is being pressed -- specifically in the case where something outside of the toolbar had focus before the click.

## 🔦 Context

There are visual oddities interacting with the toolbar on a page that has other content (which is pretty much every page 😄)
 
## 🌍 Your Environment

I was able to reproduce this in multiple browsers -- Chrome, Edge, and Firefox

In at least some cases, the issue is worse in Chrome because the previously focused button gets the `focus-visible` styling in addition to just being focused.

### PR DESCRIPTION
**Pull request by mollykreis**
# Pull Request

## 📖 Description

Resolves #6835 

There is a bug where when clicking on an element in the toolbar focuses the `activeIndex` element before focusing the clicked element. This can lead to visual oddities. This PR is intended to resolve that bug.

### 🎫 Issues

#6835 

## 👩‍💻 Reviewer Notes

I changed the toolbar to use the `mousedown` event rather than the `click` event to update the toolbar's `activeIndex` because the `mousedown` event is fired before `focusin`, thus resolving the timing issue where `focusin` focused the previously focused element prior to `click` focusing the newly focused element.

## 📑 Test Plan

I wasn't able to write a test that failed prior to my change but passes with it because the issue was a temporary state that was passed through during a `click` operations. However, I wrote two new tests that exercise the modified code path.

I also manually tested my change in Chrome, Edge, and Firefox.

### General

- [x] I have included a change request file using `$ yarn change`
- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## ⏭ Next Steps

N/A

### CODE REVIEW

**Review by scomea — APPROVED** [review]

**Comment by mollykreis** [comment]
Could I get a second review on this PR? @EisenbergEffect @chrisdholt @nicholasrice

**Comment by chrisdholt** [comment]
> Could I get a second review on this PR? @EisenbergEffect @chrisdholt @nicholasrice

Yep, I've got it pulled down to smoke and validate. Thanks

**Comment by mollykreis** [comment]
@chrisdholt, were you able to perform the validation you wanted on this branch?

File: change/@microsoft-fast-foundation-903a969b-3de3-4090-974a-3423344c571c.json
Comment (chrisdholt) [inline change/@microsoft-fast-foundation-903a969b-3de3-4090-974a-3423344c571c.json:6]:
```suggestion
  "dependentChangeType": "prerelease"
```

**Review by chrisdholt — APPROVED** [review]
Updated `dependentChangeType` - will get this in. Thanks for the contribution!

**Pull request merged**
